src/main/java/com/fzi/externalcontrol/impl/ExternalControlInstallationNodeContribution.java

```diff
@@ -169,10 +175,35 @@ public void onOk(String value) {
   }
 
   public String getUrScriptProgram() {
-	if (urScriptProgram == "") {
-	  RequestProgram sender = new RequestProgram(getHostIP(), getCustomPort());
-	  urScriptProgram = sender.sendCommand("request_program\n");
-	}
-    return urScriptProgram;
+    return urScriptControlLoop;
+  }
+
+  public void findHeaderAndControlLoop(String script) {
+    int it = 0;
+    Boolean controlLoopFound = false;
+    String[] splitData = script.split("\n");
+
+    while (it < splitData.length) {
+      // Find the first line of the control loop
+      if (splitData[it].startsWith(" socket_open(")) {
+        controlLoopFound = true;
+        break;
+      }
+      it += 1;
+    }
+
+    if (controlLoopFound) {
+      for (int i = 0; i < splitData.length; ++i) {
+        if (i < it) {
+          urScriptHeader +=  splitData[i] + "\n";
+        }
+        else if (i >= it) {
+          urScriptControlLoop += splitData[i] + "\n";
+        }
+      }
+    }
+    else {
+      urScriptControlLoop = script;
```

Comment:
We should have a more verbose (error)-output here. If we expect the script to have a certain anchor and it is not there, the user should be notified. We should at least add a log output that the program will not be split.